test/controllers/thesis_controller_test.rb

```diff
@@ -264,6 +264,117 @@ class ThesisControllerTest < ActionDispatch::IntegrationTest
     assert_select "div[data-id=thesis_#{thesis.id}] form", count: 0
   end
 
+  # ~~~~~~~~~~~~~~~~~~~~ filters on the processing queue ~~~~~~~~~~~~~~~~~~~~~~
+  test 'queryset when no status is given' do
+    sign_in users(:admin)
+    get process_path
+
+    # We want to make sure that this queryset returned by the controller is the
```

Comment:
I feel a bit like this is sliding a bit towards testing that ActiveRecord works... which is out of scope of our app concerns. I'm not suggesting we remove them as they have already been written, but I think the second set of tests is adequately 🌈

Reply:
I think it's a little more subtle than that in that it's testing that I actually wrote the queries that I intended to write, and they are wired up to parameters in the way that I intended to wire them up. (I hear you on not testing rails internals, though.)